### ISSUE TITLE
Celery task silently fails on large document insertion

### ISSUE THREAD

**Issue by pycal** [mayan-edms/mayan-edms#206]
I'm importing documents via the API. My script hangs on a document that is 1.6gb, and no further documents are imported + no output in the celery logs.

**Comment by rosarior** [comment]
Launch the celery worker with the -l DEBUG flag to see if more information is displayed in the console. If not an error at least the last message, to know where it is freezing.

**Comment by pycal** [comment]
From celery-worker.log:

```
[2015-06-10 19:54:46,934: DEBUG/Worker-1] sending chunk (40323825): '\xc1\x00\xc2\xf7\xc1\x00\xc0\xfc\xc1\x00\xc0\xf4\xc1\x05\xc0\xc1\xc0\xc0\xc1\xc0\xfb\xc1\x06\xc0\xc1\xc1\xc0\xc1\xc0\xc1\xfe\xc0\xff\xc1\x00\xc1\xfe\xc0\xff\xc1\xff\xc0\x00\xc1\xfb\xc0\x00\xc1\xf1\xc0\x00\xbf\xfd\xc0\x00\xbf\xfe\xc0\x00\xbf\xfd\xc0\x00\xbf\xfe\xc0\x00\xbf\xfe\xc0\x03\xbf\xc0\xbf\xbf\xfd\xc0\x00\xbf\xfd\xc0\xfe\xbf\x07\xc0\xbf\xc0\xbf\xbf\xc0\xbf\xc0\xfe\xbf\xff\xc0\xee\xbf\x00\xbe\xfc\xbf\xfd\xbe\xfd\xbf\xff\xbe\x00\xbf\xfa\xbe\xfd\xbf\x00\xbe\xfd\xbf\xff\xbe\x02\xbf\xbe\xbf\xfd\xbe\x07\xbf\xbe\xbf\xbe\xbf\xbf\xbe\xbf\xfd\xbe\x00\xbf\xf0\xbe\x06\xbd\xbe\xbe\xbd\xbe\xbe\xbd\xfe\xbe\xfb\xbd\x04\xbe\xbd\xbe\xbd\xbe\xfe\xbd\xff\xbe\x02\xbd\xbe\xbe\xfe\xbd\t\xbe\xbd\xbd\xbe\xbd\xbd\xbe\xbd\xbe\xbe\xec\xbd\x02\xbc\xbd\xbd\x01\xbd\xbc\xf8\xbd\xfd\xbc\x00\xbd\xfd\xbc\xff\xbd\xff\xbc\xff\xbd\xfb\xbc\xff\xbd\xe6\xbc\x10\xbb\xbc\xbc\xbb\xbc\xbc\xbb\xbc\xbc\xbb\xbb\xbc\xbb\xbc\xbb\xbc\xbb\xfe\xbc\xfe\xbb\x04\xbc\xbb\xbc\xbb\xbb\xfe\xbc\x03\xbb\xbc\xbb\xbc\xf7\xbb\x00\xbc\xf7\xbb\x05\xba\xbb\xba\xbb\xba'
[2015-06-10 19:54:46,935: DEBUG/Worker-1] gpg: no valid OpenPGP data found.
[2015-06-10 19:54:46,935: DEBUG/Worker-1] [GNUPG:] NODATA 1
[2015-06-10 19:54:46,935: DEBUG/Worker-1] [GNUPG:] NODATA 2
[2015-06-10 19:54:46,936: DEBUG/Worker-1] gpg: decrypt_message failed: eof
[2015-06-10 19:54:46,936: DEBUG/Worker-1] closed output, 40323825 bytes sent
[2015-06-10 19:54:46,936: DEBUG/Worker-1] decrypt result: ''
[2015-06-10 19:54:47,237: DEBUG/Worker-1] document: superhero2.6E.psd
[2015-06-10 19:54:47,290: DEBUG/Worker-1] document: superhero2.6E.psd
[2015-06-10 19:54:47,368: DEBUG/Worker-1] office_converter: <converter.office_converter.OfficeConverter object at 0x7f1d88611c10>
[2015-06-10 19:54:47,537: ERROR/MainProcess] Task documents.tasks.task_new_document[bb9ec275-35fd-4aa7-8614-0103cf6cb507] raised unexpected: MagicException('cannot allocate 322590608 bytes (Cannot allocate memory)image/vnd.adobe.photoshop',)
Traceback (most recent call last):
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/documents/tasks.py", line 46, in task_new_document
    Document.objects.new_document(document_type=document_type, expand=expand, file_object=file_object, label=label, description=description, language=language, user=user)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/django/db/transaction.py", line 371, in inner
    return func(*args, **kwargs)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/documents/managers.py", line 83, in new_document
    versions_created.append(self.upload_single_document(document_type=document_type, file_object=file_object, description=description, label=label, language=language or LANGUAGE, user=user))
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/django/db/transaction.py", line 371, in inner
    return func(*args, **kwargs)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/documents/managers.py", line 91, in upload_single_document
    version = document.new_version(file_object=file_object, user=user)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/documents/models.py", line 200, in new_version
    comment=comment or '',
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/django/db/models/manager.py", line 157, in create
    return self.get_queryset().create(**kwargs)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/django/db/models/query.py", line 322, in create
    obj.save(force_insert=True, using=self.db)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/documents/models.py", line 326, in save
    self.update_page_count(save=False)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/documents/models.py", line 352, in update_page_count
    detected_pages = get_page_count(filepath)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/converter/api.py", line 107, in get_page_count
    office_converter.convert(input_filepath)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/converter/office_converter.py", line 87, in convert
    self.mimetype, self.encoding = get_mimetype(open(self.input_filepath), self.input_filepath, mimetype_only=True)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/mayan/apps/mimetype/api.py", line 24, in get_mimetype
    file_mimetype = mime.from_buffer(file_description.read())
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/magic.py", line 66, in from_buffer
    return magic_buffer(self.cookie, buf)
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/magic.py", line 215, in magic_buffer
    return _magic_buffer(cookie, buf, len(buf))
  File "/var/www/dms/dms/local/lib/python2.7/site-packages/magic.py", line 168, in errorcheck_null
    raise MagicException(err)
MagicException: cannot allocate 322590608 bytes (Cannot allocate memory)image/vnd.adobe.photoshop
[2015-06-10 19:56:21,795: DEBUG/MainProcess] pidbox received method heartbeat() [reply_to:None ticket:None]
[2015-06-10 19:56:21,796: DEBUG/MainProcess] Heartbeat requested by remote.
[2015-06-10 19:56:21,829: DEBUG/MainProcess] pidbox received method stats() [reply_to:{u'routing_key': u'5f0186f1-6342-38d5-95c6-6ed7922be8be', u'exchange': u'reply.celery.pidbox'} ticket:0805dc7b-3a47-421d-8551-e2cbfd42bc69]
[2015-06-10 19:56:21,863: DEBUG/MainProcess] pidbox received method active_queues() [reply_to:{u'routing_key': u'0beeba67-a92a-3ee4-8328-2d2be959915d', u'exchange': u'reply.celery.pidbox'} ticket:4bd5d2a2-c7e2-49be-ace8-6d39d37f4b47]
[2015-06-10 19:56:21,876: DEBUG/MainProcess] pidbox received method dump_tasks(taskinfoitems=[]) [reply_to:{u'routing_key': u'e4a5105c-d7de-3ecc-96b5-5d0e5ac29e2d', u'exchange': u'reply.celery.pidbox'} ticket:df628206-5689-4c99-8aea-6eba5b4d7aa9]
[2015-06-10 19:56:21,899: DEBUG/MainProcess] pidbox received method dump_schedule(safe=False) [reply_to:{u'routing_key': u'f99d3dc1-51ea-3d36-98d6-5db092037306', u'exchange': u'reply.celery.pidbox'} ticket:a252bd61-f8d2-4bb6-813d-89abb6c7f129]
[2015-06-10 19:56:22,915: DEBUG/MainProcess] pidbox received method dump_active(safe=False) [reply_to:{u'routing_key': u'5f0186f1-6342-38d5-95c6-6ed7922be8be', u'exchange': u'reply.celery.pidbox'} ticket:66356a33-8f37-4a95-ba8e-a87ac487664a]
[2015-06-10 19:56:22,924: DEBUG/MainProcess] pidbox received method dump_conf(with_defaults=False) [reply_to:{u'routing_key': u'f99d3dc1-51ea-3d36-98d6-5db092037306', u'exchange': u'reply.celery.pidbox'} ticket:6dfadd6f-3bb1-444d-ab37-8176ffc945e3]
[2015-06-10 19:56:23,253: DEBUG/MainProcess] pidbox received method dump_revoked() [reply_to:{u'routing_key': u'0beeba67-a92a-3ee4-8328-2d2be959915d', u'exchange': u'reply.celery.pidbox'} ticket:9b77e800-2b8e-40b5-b8ec-7243ea87d2df]
[2015-06-10 19:56:23,255: DEBUG/MainProcess] pidbox received method dump_reserved(safe=False) [reply_to:{u'routing_key': u'e4a5105c-d7de-3ecc-96b5-5d0e5ac29e2d', u'exchange': u'reply.celery.pidbox'} ticket:11b5cb5e-2908-423f-9f68-68f551306234]
```

And from celery flower: https://www.dropbox.com/s/wqa6tjucc2zkzu7/Screenshot%202015-06-10%2016.06.57.png?dl=0
